language/libra-tools/transaction-replay/src/lib.rs

```diff
@@ -155,11 +157,12 @@ impl LibraDebugger {
         sender: AccountAddress,
         begin: Version,
         end: Version,
+        override_changeset: Option<MoveChanges>,
     ) -> Result<Option<Version>> {
         let predicate = compile_move_script(code_path, sender)?;
         let gas_table = zero_cost_schedule();
         let is_version_ok = |version| {
-            self.run_session_at_version(version, |session| {
+            self.run_session_at_version(version, override_changeset.clone(), |session| {
```

Comment:
`clone` needed?

Reply:
We actually need this clone as the function signature requires an `Fn` rather than `FnOnce`. You could imagine that this function will be run against multiple versions. Each run will require a copy.